### PR TITLE
Add delegate to PageIterator to allow consumers to configure next page request

### LINKED ISSUE #852: Allow consumer to configure the next page request when iterating collections
**Page Iterator Task**
From the [Page Iterator spec](https://github.com/microsoftgraph/msgraph-sdk-design/blob/master/tasks/PageIteratorTask.md)

> Accept or provide the ability to pass information like headers, middleware options in the request for the next page. For example, when using a PageIteratorTask for processing a large number of events, consumer should be able to specify the timezone preferance in the headers.

### PR DESCRIPTION
**Pull request by pschaeflein**
Fixes #852

### Changes proposed in this pull request
- Add a delegate to PageIterator the allows consumers to configure the next page request ( `IBaseRequest` ) before it is invoked.

### CODE REVIEW

**Review by zengin — DISMISSED** [review]
LGTM.

@MIchaelMainer, I think PageIterator needs to go to Core library so that Beta library can also benefit from this.

I see an issue related to PageIterator in the core repo (https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/issues/54), do we have an existing task for the migration itself?

File: src/Microsoft.Graph/Tasks/PageIterator.cs [hunk 1]
```diff
@@ -61,6 +62,21 @@ public static PageIterator<TEntity> CreatePageIterator(IBaseClient client, IColl
             };
         }
 
+        /// <summary>
```
Comment (baywet) [inline src/Microsoft.Graph/Tasks/PageIterator.cs:65]: I don't think you need a full overload if you make your parameter optional, just add the optional parameter to the existing method.

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for the great contribution! One minor change on my side.

**Review by baywet — APPROVED** [review]

**Review by MIchaelMainer — APPROVED** [review]

**Comment by MIchaelMainer** [comment]
@zengin Added https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/issues/179 to track moving this into core.

**Pull request merged**